### DISCUSSION TITLE
What exactly does 'Duration' of Sample struct do?

### DISCUSSION THREAD

**Discussion by luxi78** [pion/webrtc#2014] [Q&A] [answered]
Hello guys, 

I‘ve  found many snippets like :
```go
if ivfErr = videoTrack.WriteSample(media.Sample{Data: frame, Duration: time.Second}); ivfErr != nil {
   panic(ivfErr)
}
```
in examples,but not sure the role of `Duration`. Can someone tell me the details about it? 

Thank you!

**Comment by Sean-Der** [comment, marked as answer]
Hey @luxi78 

When the media is packetized in RTP this is used the generate the Timestamp values.

This is used to generate the sample count [here](https://github.com/pion/webrtc/blob/master/track_local_static.go#L263) which is then passed into pion/rtp.

This then is used here to increment the Timestamp [here](https://github.com/pion/rtp/blob/e8906add970eeba43d123bf3bdbe4dbaf03ae578/packetizer.go#L76)

**Reply by luxi78** [reply]
Got it, thank you Sean.
